src/lists.jl

```diff
@@ -8,13 +8,23 @@ abstract type AbstractTargetedLinkedList{T,L,N} <: AbstractPairedLinkedList{T} e
     node = ListNode(list::DoublyLinkedList, data)
 
 Create a `ListNode` belonging to the specified `list`. The node contains a reference `list` to the parent list, 
-the provided `data`, a link `prev` to the preceding node, and a link `next` to the following node.
+the provided `data`, but it has no specific insertion point into `list` (see [`insertnode!`](@ref)). # On first read I wondered if it was added to the end of the list?
+
+# The former documentation was a blend between documenting the type and constructor. I do that too.
+# However, it makes sense to prioritize clarity about the constructor, since that's what you're actually documenting here.
+# I added the next item in an attempt to recover some of the lost information from above, but I'm not sure it's a great idea
+# especially if you discourage people from manipulating the fields.
+`node.prev` and `node.next` represent the previous and next nodes, respectively, of a list.
 """
 mutable struct ListNode{T,L<:AbstractDoublyLinkedList{T}} <: AbstractListNode{T,L}
     list::L
     data::T
     prev::ListNode{T,L}
     next::ListNode{T,L}
+    # Is this method really needed? (Is there ever a reason to make a node that doesn't have `data`?)
+    # For any optional field Julia will internally check whether it's defined before returning the value to you,
+    # which results in a (small) performance hit. If you never use it, then it might be better to delete
+    # it and then Julia will know that the first two fields are always defined.
```

Comment:
The idea here was to leave the `data` undefined for the "dummy" nodes at the head or tail of the list. Since the type of the data is arbitrary, is there a straightforward choice for a default value that we can use for `data`?

Reply:
Oh, good point. No, I think you're right that we can't promise that there will be a default constructor. So your approach is the right one.